### PR TITLE
fix: keepkey recovery flow cancel and close events

### LINKED ISSUE #1610: Gracefully handle not completing the KeepKey initialization flow
### Overview

The user can select the close or back buttons to exit the initialization flow, possibly by accident. This will drop them into a broken state until they choose to disconnect their wallet. This may be a non-obvious fix, especially if the user got into the state accidentally.

165401124-20b3d3ed-632a-41c6-a086-dbdc6bbf3495

### References and additional details

We'll want to update the `onClose()` model handler (and any other relevant handlers) to ensure the app is left in a happy state.

### Acceptance Criteria

If a user disconnects their KeepKey close the modal and kick them back to the connect wallet flow. Also, send the user to the wallet connection screen if they ever close (x) out of the recovery modal.

Product to confirm.

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Estimated effort

_No response_

### LINKED ISSUE #1614: Fix "Cancel" button on "Write down Recovery Sentence" modal
### Overview

Pressing "Cancel" doesn't exit the screen or do anything, though it leaves the device in an uninitialized state.

### References and additional details

- Send a `cancel()` request to the connected device, if it doesn't do this already
- Do something with UX to make this more graceful - product to confirm, as per comment below in acceptance criteria.

### Acceptance Criteria

If a user cancels during this step have them routed back to the connect wallet screen. This is the same activity if they were to cancel during any other steps in the flow. The user would need to start the recovery flow over.

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Estimated effort

_No response_

### PR DESCRIPTION
**Pull request by NeOMakinG**
## Description
- I added a new hook: `useKeepKeyCancel` used to cancel the wallet + get back to the wallet select route
- It's used by the `AwaitKeepKey` component in both `RecoverySentence` and `RecoverySentenceEntry`
- In case the user unplug the keepkey, it opens and close the modal in order to be redirected to the `SelectModal` view

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)
closes #1614
partially addresses #1610

## Risk
KeepKey flow could be buggy

## Testing
- Wipe your keepkey
- Try to recover it, click on cancel when you can, you should be redirected to the SelectModal

- Try to initialize it, click on cancel it's waiting for a keepkey button press, you should be redirected to the SelectModal view

- Try to go anywhere in the workflow, unplug the keepkey from your computer, you should be redirected to the SelectModal view

### CODE REVIEW

File: src/context/WalletProvider/KeepKey/hooks/useKeepKeyCancel.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+export const useKeepKeyCancel = () => {
+  const history = useHistory()
+  const toast = useToast()
+  const translate = useTranslate()
+  const {
+    state: { wallet },
+    dispatch,
+  } = useWallet()
+
+  const cancelWalletRequests = useCallback(async () => {
```
Comment (0xApotheosis) [inline src/context/WalletProvider/KeepKey/hooks/useKeepKeyCancel.ts:17]: Total nitpick, but as we can only ever cancel the current request, this should be named in the singular: `cancelWalletRequest`.

File: src/context/WalletProvider/KeepKey/hooks/useKeepKeyEventHandler.ts [hunk 2]
```diff
@@ -174,6 +174,11 @@ export const useKeepKeyEventHandler = (
         if (id === state.walletInfo?.deviceId) {
           dispatch({ type: WalletActions.SET_IS_CONNECTED, payload: false })
         }
+        if (modal) {
+          // Little trick to send the user back to the wallet select route
+          dispatch({ type: WalletActions.SET_WALLET_MODAL, payload: false })
+          dispatch({ type: WalletActions.SET_WALLET_MODAL, payload: true })
```
Comment (0xApotheosis) [inline src/context/WalletProvider/KeepKey/hooks/useKeepKeyEventHandler.ts:180]: This feels a bit hacky, but probably fine.

**Review by 0xApotheosis — APPROVED** [review]
Approved with one small non-blocking request.

**Review by 0xApotheosis — APPROVED** [review]
🍭

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**